tests/StreamTest.php

```diff
@@ -158,4 +158,34 @@ public function testDoesNotThrowInToString()
         $s = new NoSeekStream($s);
         $this->assertEquals('foo', (string) $s);
     }
+
+    public function testStreamReadingWithZeroLength()
+    {
+        $r = fopen('php://temp', 'r');
+        $stream = new Stream($r);
+
+        $this->assertEquals('', $stream->read(0));
+
+        $stream->close();
+    }
+
+    public function testStreamReadingWithNegativeLength()
```

Comment:
`try..finally` will not pass checks for 5.4
I can add `expectedException` and replace code in test with
```php
try {
    $stream->read(-1);
} catch (\Exception $e) {
    $stream->close();
    throw $e;
}
$stream->close();
```